### PR TITLE
[UN-25] Handle protobuf payloads in argo-events - gcp pubsub gateway b64 encodes payload

### PR DESCRIPTION
**Pull request by amarkovits-1C**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -33,6 +78,28 @@ ifdef IMAGE_NAMESPACE
 IMAGE_PREFIX=${IMAGE_NAMESPACE}/
 endif
 
+.PHONY: build-pubsub-onec
+## Build gcp-pubsub-gateway image
+build-pubsub-onec:
+	@echo '${GREEN}building${RESET} ${YELLOW}gcp-pubsub-gateway${RESET} image'
+	@docker build \
+		--pull \
+		--build-arg github_user=$(GITHUB_USER) \
```
Comment (jakedsouza) [inline Makefile]: Do we need any of these args in the image?

File: gateways/community/gcp-pubsub/config.go [hunk 2]
```diff
@@ -31,11 +31,11 @@ type GcpPubSubEventSourceExecutor struct {
 // pubSubEventSource contains configuration to subscribe to GCP PubSub topic
 type pubSubEventSource struct {
 	// ProjectID is the unique identifier for your project on GCP
-	ProjectID string `json:"projectID"`
+	ProjectID string `yaml:"projectID"`
```
Comment (jakedsouza) [inline gateways/community/gcp-pubsub/config.go]: You should be able to use multiple tags if needed `json:"projectID", yaml:"projectID"`

File: gateways/community/gcp-pubsub/start.go [hunk 3]
```diff
@@ -59,15 +64,23 @@ func (ese *GcpPubSubEventSourceExecutor) listenEvents(ctx context.Context, sc *p
...
+	if !ok {
+		err := errors.New("topic does not exist: " + sc.Topic)
+		errorCh <- err
+		return
+	}
 
-	logger.Info("subscribing to GCP PubSub topic")
-	sub, err := client.CreateSubscription(ctx, eventSource.Id,
+	//add random name for subscription to not clash with possible existing one.
+	subName := sc.Topic + "-" + randomstring.New()
```
Comment (jakedsouza) [inline gateways/community/gcp-pubsub/start.go]: ideally ksuid, though this should be fine too, the library you are using does not seem to be maintained

File: Makefile (hunk 1 above)
Comment (amarkovits-1C) [reply Makefile]: nope, removing

File: gateways/community/gcp-pubsub/config.go (hunk 2 above)
Comment (amarkovits-1C) [reply gateways/community/gcp-pubsub/config.go]: Sounds good, adding

File: gateways/community/gcp-pubsub/start.go (hunk 3 above)
Comment (amarkovits-1C) [reply gateways/community/gcp-pubsub/start.go]: good catch, moving to ksuid

File: Makefile [hunk 4]
```diff
@@ -1,4 +1,46 @@
-PACKAGE=github.com/argoproj/argo-events
+ifndef GITHUB_USER
```
Comment (jakedsouza) [inline Makefile]: remove GITHUB_USER and GITHUB_TOKEN

Comment (amarkovits-1C) [reply Makefile]: done

**Pull request merged**